components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -91,6 +101,30 @@ void Nudging::apply_tendency(Field& base, const Field& next, const int dt)
   base.update(tend,dtend,Real(1.0));
 }
 // =========================================================================================
+void Nudging::apply_weighted_tendency(Field& base, const Field& next, const Field& weights, const int dt)
+{
+  // Calculate the weight to apply the tendency
+  const Real dtend = Real(dt)/Real(m_timescale);
+  EKAT_REQUIRE_MSG(dtend>=0,"Error! Nudging::apply_tendency - timescale tendency of " << std::to_string(dt)
+                  << " / " << std::to_string(m_timescale) << " = " << std::to_string(dtend)
+                  << " is invalid.  Please check the timescale and/or dt");
+  // Now apply the tendency.
+  Field tend = base.clone();
+
+  // Use update internal to set tendency, will be (weights*next - weights*base), note tend=base at this point.
+  auto base_view = base.get_view<const mPack**>();
```

Comment:
Since `mPack` is just a pack of size 1, why can't we just use `Real` and avoid the confusion of packs?

Reply:
Yeah, this is something that drove me crazy.  In short, the vertical interpolation function in EAMxx only works on Packs, so we cannot pass it an array of Reals.  I think the proper fix is to overhaul the vertical interpolation function to accept Real arrays.

Reply:
Yeah, but there's no interpolation here...

Reply:
Ahh right, so we should change all for of these `get_view`'s to `Real`.

Reply:
ok, that looks less confusion here